web/src/compose_fade.ts

```diff
@@ -15,57 +16,78 @@ import * as util from "./util";
 
 // TODO/TypeScript: Move this to message_list_view.js when it's migrated to TypeScript.
 export type MessageContainer = {
-    background_color: string;
-    date_divider_html?: string;
-    edited_alongside_sender: boolean;
-    edited_in_left_col: boolean;
-    edited_status_msg: boolean;
+    background_color?: string;
+    date_divider_html: string | undefined;
+    edited_alongside_sender?: boolean;
+    edited_in_left_col?: boolean;
+    edited_status_msg?: boolean;
     include_recipient: boolean;
     include_sender: boolean;
     is_hidden: boolean;
+    last_edit_timestr: string | undefined;
     mention_classname: string | null;
     message_edit_notices_in_left_col: boolean;
+    message_edit_notices_alongside_sender: boolean;
+    message_edit_notices_for_status_message: boolean;
+    modified?: boolean;
+    moved?: boolean;
```

Comment:
These two should probably be changed to pure booleans (true|false).

Reply:
I was actually flip flopping on that and ended up trying to keep the code as similar as before, but happy to add some explicit `modified: false; moved: false;` code